### PR TITLE
Extension of GraphQL API for shipment reception

### PR DESCRIPTION
**Pull request by pylipp**
Please see commits for details.

We probably need additional fields `updated_by`, `updated_on` in the `Shipment` model but I'm still clarifying with Roanna (code-wise it would be a minor change).

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 Extension of shipment GraphQL API for reception #sprint-feb-20](https://trello.com/c/SEcv4ec6/1120-20-extension-of-shipment-graphql-api-for-reception-sprint-feb-20)

File: back/boxtribute_server/business_logic/box_transfer/shipment/crud.py [hunk 1]
```diff
@@ -146,6 +146,22 @@ def send_shipment(*, id, user):
...
+    """Transition state of specified shipment to 'Receiving'.
+    Raise InvalidShipmentState exception if shipment state is different from 'Sent'.
+    """
+    shipment = Shipment.get_by_id(id)
+    if shipment.state != ShipmentState.Sent:
+        raise InvalidShipmentState(
+            expected_states=[ShipmentState.Sent], actual_state=shipment.state
+        )
+    shipment.state = ShipmentState.Receiving
+    # shipment.received_by = user.id
```
Comment (HaGuesto) [inline back/boxtribute_server/business_logic/box_transfer/shipment/crud.py:159]: what happened here @pylipp ? Is this about clarifying dates with @aerinsol ?

**Review by HaGuesto — APPROVED** [review]
love the spilt of update shipment. makes it much clearer and closer to the process.

## Code Review
looks clean, except for one comment

## functional review
- receive shipment works.
- The split of updating shipments I will have to test when we implement them. I'm now on the QrReader assigning multiple boxes to a shipment. Therefore, will have the chance to test there.

File: back/boxtribute_server/business_logic/box_transfer/shipment/crud.py (hunk 1 above)
Comment (pylipp) [reply back/boxtribute_server/business_logic/box_transfer/shipment/crud.py:159]: needs a database migration first @HaGuesto 
Then shipment table should look like:
```
  `received_on` datetime DEFAULT NULL,
  `received_by_id` int unsigned DEFAULT NULL,
  CONSTRAINT `shipment_ibfk_7` FOREIGN KEY (`received_by_id`) REFERENCES `cms_users` (`id`) ON DELETE SET NULL ON UPDATE CASCADE
```

Comment (pylipp) [reply back/boxtribute_server/business_logic/box_transfer/shipment/crud.py:159]: https://trello.com/c/omJaOa2A/1131-add-receivedon-and-receivedby-fields-to-shipments-db-table

**Pull request merged**